tests/Dialect/Polynomial/Conversions/heir_polynomial_to_llvm/lower_mul.mlir

```diff
@@ -50,87 +48,45 @@
-// CHECK:      %[[splat:.*]] = tensor.splat %[[extracted]] : tensor<2047xi32>
-// CHECK:      %[[mul:.*]] = arith.muli %cst, %[[splat]] : tensor<2047xi32>
-// CHECK:      %[[empty:.*]] = tensor.empty() : tensor<2047xi32>
+// CHECK:      %[[extracted_mul:.*]] = mod_arith.mul %[[extracted]], %[[c1_modarith]] : [[COEFF_TY]]
+// CHECK:      %[[extracted_mul_extracted:.*]] = mod_arith.extract %[[extracted_mul]]
+// CHECK:      %[[splat:.*]] = tensor.splat %[[extracted_mul_extracted]] : tensor<2047xi32>
+// CHECK:      %[[splat_enc:.*]] = mod_arith.encapsulate %[[splat]]
+// CHECK:      %[[mul:.*]] = mod_arith.mul %[[DIVISOR_MODARITH]], %[[splat_enc]]
+// CHECK:      %[[empty:.*]] = tensor.empty() : [[NAIVE_POLYMUL_TENSOR_TY]]
 // CHECK:      %[[c2047:.*]] = arith.constant 2047 : index
 // CHECK:      %[[v5:.*]] = arith.subi %[[c2047]], %[[v2]] : index
-// CHECK:      %[[extracted_slice_0:.*]] = tensor.extract_slice %[[mul]][0] [%[[v5]]] [1] : tensor<2047xi32> to tensor<?xi32>
-// CHECK:      %[[extracted_slice_1:.*]] = tensor.extract_slice %[[mul]][%[[v5]]] [%[[v2]]] [1] : tensor<2047xi32> to tensor<?xi32>
-// CHECK:      %[[inserted_slice:.*]] = tensor.insert_slice %[[extracted_slice_0]] into %[[empty]][%[[v2]]] [%[[v5]]] [1] : tensor<?xi32> into tensor<2047xi32>
-// CHECK:      %[[inserted_slice_2:.*]] = tensor.insert_slice %[[extracted_slice_1]] into %[[inserted_slice]][0] [%[[v2]]] [1] : tensor<?xi32> into tensor<2047xi32>
-// CHECK:      %[[splat_3:.*]] = tensor.splat %[[c_minus1]] : tensor<2047xi32>
-// CHECK:      %[[mul_2:.*]] = arith.muli %[[inserted_slice_2]], %[[splat_3]] : tensor<2047xi32>
-// CHECK:      %[[add:.*]] = arith.addi %[[DIVIDEND]], %[[mul_2]] : tensor<2047xi32>
-// CHECK:      scf.yield %[[add]] : tensor<2047xi32>
+// CHECK:      %[[extracted_slice_0:.*]] = tensor.extract_slice %[[mul]][0] [%[[v5]]] [1]
+// CHECK:      %[[extracted_slice_1:.*]] = tensor.extract_slice %[[mul]][%[[v5]]] [%[[v2]]] [1]
+// CHECK:      %[[inserted_slice:.*]] = tensor.insert_slice %[[extracted_slice_0]] into %[[empty]][%[[v2]]] [%[[v5]]] [1]
+// CHECK:      %[[inserted_slice_2:.*]] = tensor.insert_slice %[[extracted_slice_1]] into %[[inserted_slice]][0] [%[[v2]]] [1]
+// CHECK:      %[[add:.*]] = mod_arith.sub %[[DIVIDEND]], %[[inserted_slice_2]]
+// CHECK:      scf.yield %[[add]]
 // CHECK:    }
-// CHECK:    %[[extracted_slice:.*]] = tensor.extract_slice %[[rem_result]][0] [1024] [1] : tensor<2047xi32> to tensor<1024xi32>
-// CHECK:    return %[[extracted_slice]] : tensor<1024xi32>
+// CHECK:    %[[extracted_slice:.*]] = tensor.extract_slice %[[rem_result]][0] [1024] [1]
+// CHECK:    return %[[extracted_slice]]
 // CHECK:  }
 // CHECK: }
 
 func.func @lower_poly_mul(%poly0: !poly_ty, %poly1: !poly_ty) -> !poly_ty {
   %poly2 = polynomial.mul %poly0, %poly1 : !poly_ty
   return %poly2 : !poly_ty
 }
-
-// -----
-
-// Test a non-machine-word sized modulus to ensure it remsi's correctly
```

Comment:
Why was this test removed?

Reply:
There is no longer a remsi expected here, since the output of this pass is in mod_arith, so the mod_arith lowerings assert that remainder ops are inserted correctly. So this was a check for certain behavior that has been removed